src/devices/Ssd1351/Ssd1351.cs

```diff
@@ -0,0 +1,275 @@
+        /// </summary>
+        /// <param name="spiDevice">The SPI device used for communication.</param>
+        /// <param name="gpioController">The GPIO controller used for communication and controls the the <paramref name="resetPin"/> and the <paramref name="dataCommandPin"/>.</param>
+        /// <param name="dataCommandPin">The id of the GPIO pin used to control the DC line (data/command).</param>
+        /// <param name="resetPin">The id of the GPIO pin used to control the /RESET line (data/command).</param>
+        /// <param name="spiBufferSize">The size of the SPI buffer. If data larger than the buffer is sent then it is split up into multiple transmissions. The default value is 4096.</param>
+        public Ssd1351(SpiDevice spiDevice, int dataCommandPin, int resetPin, int spiBufferSize = DefaultSPIBufferSize, IGpioController gpioController = null)
+        {
+            if (!InRange((uint)spiBufferSize, 0x1000, 0x10000))
+            {
+                throw new ArgumentException($"SPI Buffer Size must be between 4096 and 65536.", nameof(spiBufferSize));
+            }
+
+            _gpioDevice = gpioController;
+
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+
+
+            _dcPinId = dataCommandPin;
+            _resetPinId = resetPin;
+
+            if(_gpioDevice == null)
+            {
+                _gpioDevice = new GpioController();
+                _disposeGpioController = true;
+            }
+
+            _gpioDevice.OpenPin(_dcPinId, PinMode.Output);
+            _gpioDevice.OpenPin(_resetPinId, PinMode.Output);
+
+            _spiBufferSize = spiBufferSize;
+
+            SetSegmentReMapColorDepth();
+        }
+
+        /// <summary>
+        /// Convert a color structure to a byte tuple represening the colour in 565 format.
+        /// </summary>
+        /// <param name="color">The color to be converted.</param>
+        private (byte, byte) Color565(Color color)
+        {
+            // get the top 5 MSB of the blue or red value
+            UInt16 retval = (UInt16)((_colorSequence == ColorSequence.BGR ? color.B : color.R) >> 3);
+            // shift right to make room for the green Value
+            retval <<= 6;
+            // combine with the 6 MSB if the green value
+            retval |= (UInt16)(color.G >> 2);
+            // shift right to make room for the red or blue Value
+            retval <<= 5;
+            // combine with the 6 MSB if the red or blue value
+            retval |= (UInt16)((_colorSequence == ColorSequence.BGR ? color.R : color.B) >> 3);
+            return ((byte)(retval >> 8), (byte)(retval & 0xFF));
+        }
+
+        /// <summary>
+        /// Send filled rectangle to the ssd1351 display.
+        /// </summary>
+        /// <param name="color">The color to fill the rectangle with.</param>
+        /// <param name="x">The x co-ordinate of the point to start the rectangle at in pixels (0..126 for the Ssd1351).</param>
```

Comment:
Can we add a quick remarks comment that mentions which corner is x = 0 and y =0 ? I assume bottom left?

Reply:
actually the origin is top left. I have put this information in the param tag.